62.CAD/vertex_shader.hlsl

```diff
@@ -85,35 +280,263 @@ PSInput main(uint vertexID : SV_VertexID)
-        // So we will decide the position of the vertex (in ndc space) based on:
-            // vertexId + sectionIdx + antiAliasedLineWidth + transformedPoints
-        // make sure to work in screen space first (because antiAliasedLineWidth and transformedPoints are in screenSpace) and then transform back to ndc
-            // see how we generate a single cage for the line above
-        // We will split Quadratic Beziers into 3 sections always.
-        // The reason we do this in vertex shader (do calc in screen and transform back to ndc) is because the lineWidth can have fixed size in screenspace
-            // imagine zooming in/out but the line width is 2 pixels always
-            // the vertices should obviously move outwards in screen when we zoom in, but they shouldn't get bigger so we can't simply transform them by applying a simple projection matrix like we do in graphics 101
-            // so vertex positioning is dependant on the projection matrix and we do this here instead of moving vertices each frame on the cpu and then uploading them to gpu
-        if (vertexIdx == 0u)
-            outV.position = float4(-1, -1, 0, 1);
-        else if (vertexIdx == 1u)
-            outV.position = float4(-1, +1, 0, 1);
-        else if (vertexIdx == 2u)
-            outV.position = float4(+1, -1, 0, 1);
-        else if (vertexIdx == 3u)
-            outV.position = float4(+1, +1, 0, 1);
+        float2 vectorAB = transformedPoints[1u] - transformedPoints[0u];
+        float2 vectorAC = transformedPoints[2u] - transformedPoints[1u];
+
+        // The area of the triangle is half of the absolute value of the cross product
+       float area = abs(vectorAB.x * vectorAC.y - vectorAB.y * vectorAC.x) * 0.5;
+       float MaxCurve;
+       if (length(transformedPoints[1u] - C1) > Radius && length(transformedPoints[1u] - C2) > Radius) {
+           MaxCurve = pow(length(transformedPoints[1u] - Mid), 3) / (area * area);
+       } else {
+           MaxCurve = max(area / pow(length(transformedPoints[0u] - transformedPoints[1u]), 3), area / pow(length(transformedPoints[2u] - transformedPoints[1u]), 3));
+       }
+       if (MaxCurve * screenSpaceLineWidth / 2.0f > 0.5f) {
+           float4 Pos0;
+           float4 Pos1;
+           if (subsectionIdx == 0 && bezierOBB_PCA(transformedPoints[0u], transformedPoints[1u], transformedPoints[2u], Pos0, Pos1, screenSpaceLineWidth / 2.0f)) {
+               if (vertexIdx == 0u)
+                   outV.position = float4(Pos0.xy, 0.0, 1.0f);
+               else if (vertexIdx == 1u)
+                   outV.position = float4(Pos0.zw, 0.0, 1.0f);
+               else if (vertexIdx == 2u)
+                   outV.position = float4(Pos1.zw, 0.0, 1.0f);
+               else if (vertexIdx == 3u)
+                   outV.position = float4(Pos1.xy, 0.0, 1.0f);
+           } else {
+               outV.position = float4(0, 0, 0.0, 1.0f);
+           }
+       }
+       else {
+
+#if 1
```

Comment:
instead of `#if 1` do `#if METHOD_XXX #elif METHOD_YYY`

then add this above in the start of the vertex shader:
```
#define METHOD_XXX // Brief Description of Method XXX
#define METHOD_YYY // Brief Description of Method YYY
```
Obviously name the methods instead of XXX :D